File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/idscp_core/Idscp2ConnectionListener.java
package de.fhg.aisec.ids.idscp2.idscp_core;

/**
 * An IDSCP Connection Listener interface that is implemented by the IDSCP2 Server to notify the
 * server about lifetimes of IDSCP connections. The server caches all active connections.
 */
public interface Idscp2ConnectionListener {

    void onError(String error);

    void onClose(String connectionId);
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/idscp_core/Idscp2ConnectionListener.kt
package de.fhg.aisec.ids.idscp2.idscp_core

/**
 * An IDSCP Connection Listener interface that is implemented by the IDSCP2 Server to notify the
 * server about lifetimes of IDSCP connections. The server caches all active connections.
 */
interface Idscp2ConnectionListener {
    fun onError(t: Throwable)
    fun onClose(connection: Idscp2Connection)
}